x/ibc/handler.go

```diff
@@ -37,22 +38,75 @@ func NewHandler(k Keeper) sdk.Handler {
 
 		// IBC channel msgs
 		case channel.MsgChannelOpenInit:
-			res, cap, err := channel.HandleMsgChannelOpenInit(ctx, k.ChannelKeeper, msg)
+			cbs := k.PortKeeper.LookupModule(ctx, msg.PortID)
+			portCap, _ := cbs.GetCapability(ctx, ibctypes.PortPath(msg.PortID))
+			err := cbs.OnChanOpenInit(ctx, msg.Channel.Ordering, msg.Channel.ConnectionHops, msg.PortID, msg.ChannelID, msg.Channel.Counterparty, msg.Channel.Version)
```

Comment:
Call user defined `OnChanOpenInit` callback

Reply:
Hmm, should this happen before or after `HandleMsgChannelOpenInit`?

Both are safe as long as the transaction is atomic. If it's after then it can expect the channel state to be there.